daemon/daemon.go

```diff
@@ -874,7 +894,19 @@ func (daemon *Daemon) Shutdown() error {
 				return
 			}
 			logrus.Debugf("stopping %s", c.ID)
-			if err := daemon.shutdownContainer(c); err != nil {
+
+			// temporarily disable AutoRemove
+			autoRemove := c.HostConfig.AutoRemove
+
+			c.HostConfig.AutoRemove = false
+			err := daemon.shutdownContainer(c)
+			// restore AutoRemove irrespective of whether the shutdown worked or not
+			c.HostConfig.AutoRemove = autoRemove
+			if toDiskErr := c.ToDiskLocking(); toDiskErr != nil {
```

Comment:
Never mind - now I see the comment you added below: https://github.com/docker/docker/pull/20848/files#r56016989